java/vector/src/test/java/org/apache/arrow/vector/TestSplitAndTransfer.java

```diff
@@ -198,6 +202,47 @@ public void testWithEmptyVector() {
     toDUV.clear();
   }
 
+  @Test
+  public void testListVectorWithEmptyMapVector() {
+    // List<element: Map(false)<entries: Struct<key: Utf8 not null, value: Utf8> not null>>
+    int valueCount = 1;
+    List<Field> children = new ArrayList<>();
+    children.add(new Field("key", FieldType.notNullable(new ArrowType.Utf8()), null));
+    children.add(new Field("value", FieldType.nullable(new ArrowType.Utf8()), null));
+    Field structField = new Field(
+            "entries",
+            FieldType.notNullable(ArrowType.Struct.INSTANCE),
+            children);
+
+    Field mapField = new Field(
+            "element",
+            FieldType.notNullable(new ArrowType.Map(false)),
+            asList(structField));
+
+    Field listField = new Field(
+            "list",
+            FieldType.nullable(new ArrowType.List()),
+            asList(mapField));
+
+    ListVector fromListVector = (ListVector) listField.createVector(allocator);
+    fromListVector.allocateNew();
+    fromListVector.setValueCount(valueCount);
+
+    // child vector is empty
+    MapVector dataVector = (MapVector) fromListVector.getDataVector();
+    dataVector.allocateNew();
+    // unset capacity to mimic observed failure mode
+    dataVector.getOffsetBuffer().capacity(0);
+
+    TransferPair transferPair = fromListVector.getTransferPair(fromListVector.getAllocator());
+    transferPair.splitAndTransfer(0, valueCount);
+    ListVector toListVector = (ListVector) transferPair.getTo();
+
+    assertEquals(valueCount, toListVector.getValueCount());
+    fromListVector.clear();
```

Comment:
nit: use try-with-resources

Reply:
Note that there're several closeable resources here and `toListVector` cannot be constructed inside the same `try` as `fromListVector`. This is just to say that there'd be some ugly nesting if using try.

Reply:
Right. Sure.